File: 3702-longest-subsequence-with-non-zero-bitwise-xor/3702-longest-subsequence-with-non-zero-bitwise-xor.go
func longestSubsequence(nums []int) int {
	xorValue := 0
	hasNonZero := false

	for _, x := range nums {
		xorValue ^= x

		if x != 0 {
			hasNonZero = true
		}
	}

	if xorValue != 0 {
		return len(nums)
	}

	if hasNonZero {
		return len(nums) - 1
	}

	return 0
}

File: 3702-longest-subsequence-with-non-zero-bitwise-xor/3702-longest-subsequence-with-non-zero-bitwise-xor.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestSubsequence = function(nums) {
    let xorValue = 0;
    let hasNonZero = false;

    for (const x of nums) {
        xorValue ^= x;

        if (x !== 0) {
            hasNonZero = true;
        }
    }

    if (xorValue !== 0) {
        return nums.length;
    }

    if (hasNonZero) {
        return nums.length - 1;
    }

    return 0;
};